pkg/alertmanager/state_replication_test.go

```diff
@@ -161,47 +175,32 @@ func TestStateReplication(t *testing.T) {
 			require.Eventually(t, func() bool {
 				replicator.mtx.Lock()
 				defer replicator.mtx.Unlock()
-				return len(replicator.results) == len(tt.results)
+				return len(replicator.results) == len(tt.replicationResults)
 			}, time.Second, time.Millisecond)
 
 			if tt.replicationFactor > 1 {
+				// If the replication factor is greater than 1, we expect state to be loaded from other Alertmanagers
 				assert.NoError(t, testutil.GatherAndCompare(reg, strings.NewReader(`
-# HELP alertmanager_state_fetch_replica_state_failed_total Number of times we have failed to read and merge the full state from another replica.
```

Comment:
Why have been removed? It shouldn't change anything if RF>1.

Reply:
I removed this because it was noisy and the most important metric is `alertmanager_state_initial_sync_completed_total` since it more directly represents what this method is testing: what the result of trying to load initial state is.